python/lsst/meas/algorithms/astrometrySourceSelector.py

```diff
@@ -171,6 +176,14 @@ def _isUsable(self, sourceCat):
             & self._goodSN(sourceCat) \
             & ~sourceCat.get(self.fluxFlagKey)
 
+    def _isPrimary(self, sourceCat):
+        """Return True if this is a primary source.
+        """
+        if self.primaryKey:
+            return sourceCat.get(self.primaryKey)
+        else:
+            return np.ones(len(sourceCat), dtype=bool)
+
```

Comment:
Seeing this, I'm wondering if the same should be added to `ObjectSizeStarSelectorTask` too?

Reply:
To the best of my knowledge, that selector is never run on catalogs that have had sky sources added; it's only run at the earliest time when we're looking for stars to measure the PSF.

Reply:
Yeah, I was just thinking that, in principle, someone *could*, so it wouldn’t hurt…your call!